frontend/src/metabase-types/store/state.ts

```diff
@@ -36,7 +36,7 @@ export interface State {
   modal: modalName;
 }
 
-export type Dispatch<T = any> = (action: T) => void;
+export type Dispatch<T = any> = (action: T) => unknown | Promise<unknown>;
```

Comment:
shoudln't it be like `export type Dispatch = typeof store.dispatch`?

Reply:
I'd like that, but:
- `redux`'s `Dispatch` does not support promises
- `redux-promise`'s  override for `Dispatch` does not seem to kick in 
- `redux-promise` does not export their `Dispatch` type